concent_api/core/validation.py

```diff
@@ -100,7 +134,8 @@ def validate_all_messages_identical(golem_messages_list: List[message.Message]):
                         slot,
                         getattr(base_golem_message, slot),
                         getattr(golem_message, slot),
-                    )
+                    ),
+                    error_code=ErrorCode.MESSAGE_WRONG_FIELDS,
```

Comment:
This is technically correct but misleading in my opinion. The code should indicate that it's not a problem with data format. It's that the messages are expected to be identical but they are not.

Also, I think it's the wrong place to construct this error because this function does not know enough context about why there's a problem and why it's a problem. It should just return a value indicating that messages are different and the caller should check it and construct the error message himself using his broader knowledge about the situation. By trying to be this generic you make the error incomprehensible to the client. The client may not be aware that you're putting messages on some list, where those messages come from and which one has which index. This is an implementation detail. The client should just be told which field does not match which other field.

Reply:
:+1: I will change to `MESSAGES_NOT_IDENTICAL`.

About the second part, lets create the separate issue out of it for two reasons; first is that this pull requests is going to be big even without those changes (tests updates are still to come), second is that it is not related to adding error code. If you agree I will add it.

Reply:
OK. Let's do it separately. Could you create the issue?

Reply:
Yes, I will add it.

Reply:
@cameel https://github.com/golemfactory/concent/issues/394